doc/ztp/ztp_hld.md

```diff
@@ -0,0 +1,155 @@
+# Zero Touch Provisioning (ZTP)
+# High Level Design
+
+### Rev 1.0
+
+# Table of Contents
+  * [Revision](#revision)
+  * [Requirements](#requirements)
+  * [Flow](#flow)
+  * [Error Cases](#error-cases)
+  * [ZTP CLI](#ztp-cli)
+  * [Action Items](#action-items)
+  * [Phases](#phases)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 |   11/3/18   |    Simone Salman   | Initial version                   |
+
+# Requirements
+- Build option to have ZTP enabled
+- SNMP should be available through the ZTP service
+- ZTP should verify, download and install SONiC image
+- Updategraph handles all configuration during ZTP service
+- ZTP should receive post config validation script, that collects switch info and sends it back to a remote location for processing
+- ZTP status may be logged through syslog
+- Interruption of ZTP service should disable the service gracefully
+
+# Flow
+![](https://github.com/simone-dell/SONiC/blob/ztp/images/ztp_hld/ztp_flow.jpg)
+
+For ZTP using DHCP, provisioning initially takes place over the management network and is initiated through a DHCP hook. A DHCP option is used to specify a configuration script. This script is then requested from the Web server and executed locally on the switch.
+1.	Simplify installation of switch, the steps involved will be
+    - Rack and Stack 
+    - Connect 
+    - Power-on 
+2.	Onie Boots into SONiC (with ZTP enabled) from flash
+    - Power-on ZTP service will take over to load the initial configuration.
+    - ZTP service flag $enabled is true
+    - ZTP service flag $post_install is false
+    - ZTP service flag $post_config is false
+3.	The switch now enters ZTP mode and does the following,
+    - Obtains an IP address from the DHCP server
+    - Obtains the URL for the SONIC image, provisioning script, and post config validation script 
+4.	At this step, the switch will have information to reach the HTTP / TFTP server information by DHCP option 66 or Option 240.
+5.	ZTP service will relay to updategraph service
+    - Within updategraph, ztp_enabled is true, and post_install is false
+    - This triggers updategraph to configure the device with an default config
+    - Upon exiting updategraph, SNMP will start
+6.	ZTP service will now download the software image file.
+7.	ZTP service will validate image to be a SONIC image compatible with the current platform.
+8.	ZTP service will check remaining hard disk for space before doing image install.
```

Comment:
step 8 should be sonic_installer function. does not belong to ztp service. please add this to sonic_installer.